# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/BodyStream.h
# Commit: dbc816fbc0d1
# Full Hash: dbc816fbc0d1a17cfdbb9d3ec97d3d31188323b3
# Author: Kagami Sascha Rosylight <krosylight@mozilla.com>
# Date: 2023-02-18 04:50:51
# Regressor Bug: 1813011
# File Overlap Count: 1
# Description:
#   Bug 1813011 - Part 3: Remove BodyStream::mState r=smaug
#   
#   1. eInitializing is only used to call MarkAsRead, but it can just be called unconditionally since it's just a boolean setter.
#   2. eClosed is aliased to `!mStreamHolder`, since it becomes null when being closed.
#   
# ==============================================================================

diff -r 7a10fc19c41c -r dbc816fbc0d1 dom/base/BodyStream.h
--- a/dom/base/BodyStream.h	Fri Feb 17 18:32:25 2023 +0000
+++ b/dom/base/BodyStream.h	Fri Feb 17 18:32:25 2023 +0000
@@ -134,24 +134,17 @@
 
   void ReleaseObjects();
 
-  // Common methods
-
-  enum State {
-    // This is the beginning state before any reading operation.
-    eInitializing,
+  // The closed state should ultimately be managed by the source algorithms
+  // class. See also bug 1815997.
+  bool IsClosed() { return !mStreamHolder; }
 
-    // Stream is initialized and being consumed.
-    eInitialized,
-
-    // Operation completed.
-    eClosed,
-  };
-
-  State mState;
+  // Common methods
 
   // mGlobal is set on creation, and isn't modified off the owning thread.
   // It isn't set to nullptr until ReleaseObjects() runs.
   nsCOMPtr<nsIGlobalObject> mGlobal;
+  // Same for mStreamHolder. mStreamHolder being nullptr means the stream is
+  // closed.
   RefPtr<BodyStreamHolder> mStreamHolder;
   nsCOMPtr<nsIEventTarget> mOwningEventTarget;
   // Same as mGlobal but set to nullptr on OnInputStreamReady (on the owning
